ComunityMembers/RepositoryTemplate/AbstractRepository.cs

```diff
@@ -0,0 +1,60 @@
+﻿using CommunityMembersModels.Contracts;
+using CommunityRepository.DatabaseContext;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace CommunityRepository
+{
+    public abstract class AbstractRepository<T> where T : IModel
+    {
+        public FullDbContext _dataContext { get; set; }
+
+        public AbstractRepository(FullDbContext fullContext)
+        {
+            _dataContext = fullContext;
+        }
+        public void Delete(int Id)
```

Comment:
Since it uses the `GetAsync` and `DeleteAsync` methods this method must be declared as `async` 

```
public async Task Delete(int id)
{
   {
            T item = await GetAsync(Id);
            if(item != null)
            {
                await DeleteAsync(item);
            }
        }
}
```